src/core/Akka.FSharp/Spawning.fs

```diff
@@ -12,129 +12,135 @@ open Microsoft.FSharp.Quotations
+module System =
     /// Creates an actor system with remote deployment serialization enabled.
-    let create (name : string) (config : Akka.Configuration.Config) : ActorSystem = 
+    let create (name : string) (config : Akka.Configuration.Config) : ActorSystem =
         let system = ActorSystem.Create(name, config)
         Serialization.exprSerializationSupport system
         system
 
 [<AutoOpen>]
-module Spawn = 
-    type SpawnOption = 
+module Spawn =
+    type SpawnOption =
         | Deploy of Deploy
         | Router of Akka.Routing.RouterConfig
         | SupervisorStrategy of SupervisorStrategy
         | Dispatcher of string
         | Mailbox of string
-    
-    let rec applySpawnOptions (props : Props) (opt : SpawnOption list) : Props = 
+
+    let rec applySpawnOptions opt (props : Props) =
         match opt with
         | [] -> props
-        | h :: t -> 
-            let p = 
+        | h :: t ->
+            let p =
                 match h with
                 | Deploy d -> props.WithDeploy d
                 | Router r -> props.WithRouter r
                 | SupervisorStrategy s -> props.WithSupervisorStrategy s
                 | Dispatcher d -> props.WithDispatcher d
                 | Mailbox m -> props.WithMailbox m
-            applySpawnOptions p t
-    
+            applySpawnOptions t p
+
+    let spawnFromProps (actorFactory : IActorRefFactory) name props =
+        actorFactory.ActorOf(props, name)
+        |> typed
+        :> ActorRef<'Message>
+
+    let createFromExpression actorFactory options name : Linq.Expressions.Expression<Func<'TActor>> -> ActorRef<'Message>=
+        Props.Create
+        >> applySpawnOptions options
+        >> spawnFromProps actorFactory name
+
     /// <summary>
     /// Spawns an actor using specified actor computation expression, using an Expression AST.
     /// The actor code can be deployed remotely.
     /// </summary>
     /// <param name="actorFactory">Either actor system or parent actor</param>
     /// <param name="name">Name of spawned child actor</param>
-    /// <param name="expr">F# expression compiled down to receive function used by actor for response for incoming request</param>
+    /// <param name="actorExpr">F# expression compiled down to receive function used by actor for response for incoming request</param>
     /// <param name="options">List of options used to configure actor creation</param>
-    let spawne (actorFactory : IActorRefFactory) (name : string) 
-        (expr : Expr<Actor<'Message> -> Cont<'Message, 'Returned>>) (options : SpawnOption list) : ActorRef<'Message> = 
-        let e = Linq.Expression.ToExpression(fun () -> new FunActor<'Message, 'Returned>(expr))
-        let props = applySpawnOptions (Props.Create e) options
-        typed (actorFactory.ActorOf(props, name)) :> ActorRef<'Message>
-    
+    let spawne actorFactory options name (actorExpr : Expr<Actor<'Message> -> Cont<'Message, 'Returned>>) : ActorRef<'Message> =
```

Comment:
The Akka.FSharp convention, we follow is to explicitly type each parameter used in public functions/methods. This way we have better control over incoming changes and reduce the risk of accidental breaking the API.